src/coreclr/src/jit/compiler.cpp

```diff
@@ -4624,6 +4625,12 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
         compCurBB = nullptr;
 #endif // DEBUG
 
+        // If we needed to create any new BasicBlocks then renumber the blocks
+        if (fgBBcount > prevBBCount)
```

Comment:
What if we delete a block? Should we renumber then? I guess in that case the assert you're fixing wouldn't be affected, but there could be "holes" in the block list. Would it be safer to just always call `fgRenumberBlocks`, at the cost of iterating through the block list?

Reply:
There isn't a problem with deleting blocks, this is necessary because some phases in the Jit make a simplifying assumption that blocks are numbered in an ascending order. 

These two cases are seemingly rare right now as only the JitStress case (for GS Cookies (and then only sometimes))
and likewise rare for the Global Morph case.

We would like each phase that comes after the dominators computation to keep this invariant true.  I don't think that we should have every phase call fgRenumberBlocks and when it is rare we should only call it when necessary.

Reply:
It might makes sense to document (as comment) why we don't need to just call `fgRenumberBlocks` and why the check used is sufficient.

You could imagine code (maybe we already have it?) that verifies that blocks are properly renumbered and that `fgRenumberBlocks` would do nothing, and then write here:
```
if (fgBBcount > prevBBCount)
{
  fgRenumberBlocks();
}
else
{
  assert(fgBlocksDoNotNeedToBeRenumbered());
}
```